packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,155 @@
+}
+
+/// Service for setting the content sensitivity of the native app window (Android `View`)
+/// that contains the app's widget tree.
+///
+/// This service is only currently supported on Android API 35+.
+class SensitiveContentService {
+  /// Creates service to set content sensitivity of an app window (Android `View`) via
+  /// communication over the sensitive content [MethodChannel].
+  SensitiveContentService() {
+    sensitiveContentChannel = SystemChannels.sensitiveContent;
+  }
+
+  /// The channel used to communicate with the shell side to get and set the
+  /// content sensitivity of an app window (Android `View`).
+  late MethodChannel sensitiveContentChannel;
+
+  /// Sets content sensitivity level of the app window (Android `View`) that contains the app's widget
+  /// tree to the level specified by [contentSensitivity] via a call to the native embedder.
+  Future<void> setContentSensitivity(ContentSensitivity contentSensitivity) async {
+    try {
+      await sensitiveContentChannel.invokeMethod<void>(
+        'SensitiveContent.setContentSensitivity',
+        contentSensitivity.name,
+      );
+    } catch (e) {
+      // Content sensitivity failed to be set.
+      throw FlutterError('Content sensitivity $contentSensitivity failed to be set: $e');
+    }
+  }
+
+  /// Gets content sensitivity level of the app window (Android `View`) that contains
+  /// the app's widget tree.
+  Future<ContentSensitivity> getContentSensitivity() async {
+    String? result;
+    try {
+      result = await sensitiveContentChannel.invokeMethod<String>(
+        'SensitiveContent.getContentSensitivity',
+      );
+    } catch (e) {
+      // Content sensitivity failed to be retrieved.
+      throw FlutterError('Failed to retrieve content sensitivity: $e');
+    }
+
+    final ContentSensitivity contentSensitivity = ContentSensitivity.values.firstWhere(
+      (ContentSensitivity cs) => cs.name == result && cs != ContentSensitivity._unknown,
+      orElse:
+          () =>
+              throw UnsupportedError(
+                'Android Flutter View has a content sensitivity mode '
+                'that is not recognized by Flutter. If you see this error, it '
+                'is possible that the View uses a new mode that Flutter needs to '
+                'support; please file an issue.',
+              ),
+    );
+
+    return contentSensitivity;
+  }
+
+  /// Returns whether or not setting content sensitivity levels is supported
+  /// by the device.
+  ///
+  /// This method must be called before attempting to call [getContentSensitivity]
+  /// or [setContentSensitivity].
+  ///
+  /// This feature is only supported on Android 35+ currently. Its return value will
+  /// not change and thus, is safe cache.
+  Future<bool> isSupported() async {
+    if (defaultTargetPlatform != TargetPlatform.android) {
+      return false;
+    }
+    return (await sensitiveContentChannel.invokeMethod<bool>('SensitiveContent.isSupported'))!;
```

Comment:
Why the `!` here?

Reply:
The method channel call results in `Future<bool>` which according to the analyzer, can be null.

Reply:
Oh right. This is probably another thing that we should do error handling for. Currently it looks like there is no try/catch where this is called in _register. Currently if this call fails, or if it unexpectedly returns null, then the app will crash. Instead I guess we should just consider it to be unsupported?

Reply:
Sure, changed!